recipes/miniaudio/all/conanfile.py

```diff
@@ -0,0 +1,37 @@
+from conans import ConanFile, tools
+import os
+import glob
+
+
+class MiniaudioConan(ConanFile):
+    name = "miniaudio"
+    description = "A single file audio playback and capture library."
+    topics = ("conan", "miniaudio", "header-only", "sound")
+    homepage = "https://github.com/mackron/miniaudio"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Unlicense"
+    settings = "os", "compiler", "build_type", "arch"
+
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob(self.name + "-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="miniaudio.h", dst="include", src=self._source_subfolder)
+
+    def package_info(self):
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.extend(["m", "pthread", "dl"])
+        elif self.settings.os == "Macos":
+            self.cpp_info.system_libs.extend(["m", "pthread"])
```

Comment:
```suggestion
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.extend(["m", "pthread"])
        if self.settings.os == "Linux":
            self.cpp_info.system_libs.append("dl")
```

https://github.com/mackron/miniaudio#building

Reply:
FreeBSD will include Macos?

Reply:
I included Macos because:
"The macOS build should compile cleanly without the need to download any dependencies nor link to any libraries or frameworks. The iOS build needs to be
compiled as Objective-C and will need to link the relevant frameworks but should compile cleanly out of the box with Xcode. Compiling through the command line
requires linking to `-lpthread` and `-lm`."

Reply:
Link to this statement? Maybe some frameworks are needed, but not pthread or m.

Reply:
That statement is in the header file itself. It contains basically the full documentation, including building:
https://raw.githubusercontent.com/mackron/miniaudio/master/miniaudio.h

Reply:
oh yes, forgot the design C header only lib with macro implementation, always a nightmare to package.

Reply:
Yeah, it's a bit weird. Also, speaking of that, a macro changes the licensing. Any ideas for that?

Reply:
There is also this statement:

```
Due to the way miniaudio links to frameworks at runtime, your application may not pass Apple's notarization process. To fix this there are two options. The
first is to use the `MA_NO_RUNTIME_LINKING` option, like so:

    ```c
    #ifdef __APPLE__
        #define MA_NO_RUNTIME_LINKING
    #endif
    #define MINIAUDIO_IMPLEMENTATION
    #include "miniaudio.h"
    ```

This will require linking with `-framework CoreFoundation -framework CoreAudio -framework AudioUnit`.
```

Reply:
Maybe add `CoreFoundation`, `CoreAudio` and `AudioUnit` to `cpp_info.frameworks` for macos, and force `MA_NO_RUNTIME_LINKING` in `cpp_info.defines`

Reply:
```
The Speex resampler is made up of third party code which is released under the BSD license. Because it is licensed differently to miniaudio, which is public
domain, it is strictly opt-in and all of it's code is stored in separate files. If you opt-in to the Speex resampler you must consider the license text in it's
source files. To opt-in, you must first `#include` the following file before the implementation of miniaudio.h:

#include "extras/speex_resampler/ma_speex_resampler.h"
```

I forgot this file in the recipe!